src/components/CPU/BreakpointPanel.js

```diff
@@ -0,0 +1,209 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import './BreakpointPanel.css';
+import { ContextMenuTrigger } from 'react-contextmenu';
+import listeners from '../../utils/listeners';
+import BreakpointContextMenu from './BreakpointPanelContextMenu';
+import BreakpointModal from './BreakpointModal';
+import BreakpointPanelItem from './BreakpointPanelItem';
+import { hasContextMenu } from '../../utils/dom';
+import { useDebuggerContext } from '../DebuggerContext';
+
+export default function BreakpointPanel(props) {
+	const context = useDebuggerContext();
+
+	const [cpuBreakpoints, setCpuBreakpoints] = useState([]);
+	const [memoryBreakpoints, setMemoryBreakpoints] = useState([]);
+	const [editingBreakpoint, setEditingBreakpoint] = useState(null);
+	const [creatingBreakpoint, setCreatingBreakpoint] = useState(false);
+	const [selectedRow, setSelectedRow] = useState(-1);
+
+	const breakpoints = [...memoryBreakpoints, ...cpuBreakpoints];
+
+	const fetchCpuBreakpoints = useCallback(() => {
+		if (!context.gameStatus.started) {
+			return;
+		}
+		context.ppsspp.send({
+			event: 'cpu.breakpoint.list',
+		}).then(({ breakpoints }) => {
+			setCpuBreakpoints(breakpoints);

```

Comment:
I'd prefer to decorate these with a type (i.e. `breakpoints.map(b => { ...b, type: 'cpu' })`) than assume type based on `.size` as we do later below.

This would also simplify some of the `setEditingBreakpoint` stuff.

-[Unknown]

Reply:
Yeah, this will be better solution. I'll change it even though I'm not a fan of decorating API response like that. It would be nice for breakpoint type to be included in PPSSPP response. Checks for breakpoint type are also needed in breakpoint modal.
Breakpoint modal is already using `memory` and `execute` so it would probably be better to use that instead of 'memory' and 'cpu' for consistency.